service/student_service_test.go

```diff
@@ -0,0 +1,406 @@
+package service_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"example.com/student-service/proto"
+	"example.com/student-service/repository/student"
+	"example.com/student-service/service"
+	"example.com/student-service/service/mocks"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/golang/mock/gomock"
+)
+
+type fixedIDGenerator struct{}
+
+func (f fixedIDGenerator) GenerateID() string {
+	return "generated-id"
+}
+
+func setupTest(t *testing.T) (
```

Comment:
You can use testsuites, that will help manage and share test dependencies via utils like SetupSuite, SetupTests, AfterTest, AfterSuit logic, see if you like it